augraphy/augmentations/bookbinding.py

```diff
@@ -33,16 +34,23 @@ def __repr__(self):
         return f"BookBinding(radius_range={self.radius_range}, curve_intensity_range={self.curve_intensity_range},  p={self.p})"
 
     def add_book_shadow(self, img, radius, angle):
-        img_output = img.copy()
+
         rows = img.shape[0]
         cols = img.shape[1]
-        for i in range(rows):
-            for j in range(cols):
-                dist = math.sqrt((j * j))
-                d = dist + (radius * (1 - math.cos(angle)))
-                new_i = img[i, j] * (dist / (d)) ** 2
-                img_output[i, j, :] = new_i
-        return img_output
+
+        # compute mask of shadow
+        img_dist = np.repeat(np.arange(cols), rows)
```

Comment:
In this update, i vectorized the shadow related section here to speed up the computation speed of book binding.

Previously it took 8-9 seconds in my machine and now it took only 2-3 seconds.

Reply:
Great work, this will help a lot especially in Colab notebooks